src/Symfony/Component/Routing/Route.php

```diff
@@ -453,6 +453,10 @@ public function setRequirements(array $requirements)
     public function addRequirements(array $requirements)
     {
         foreach ($requirements as $key => $regex) {
+            if (is_int($key)) {
+                @trigger_error(sprintf('Support for numeric requirements ("%d" given) is deprecated since Symfony 4.2 and will trigger an error in 5.0.'), E_USER_DEPRECATED);
```

Comment:
the sentence misses context, if I see this, I'm going to wonder hard where this comes from.

Reply:
Unfortunately, we don't have so much context (except maybe for the path) that we could provide here. Should we do this in the loader class instead where we can hint to classes/files and the names of the routes?

Reply:
I updated the `AnnotationClassLoader` and `YamlFileLoader` instead to be able include more detailed information.